Common/Statistics/AssetsUnderManagementCapacityManager.cs

```diff
@@ -0,0 +1,238 @@
+        public AssetsUnderManagementCapacityManager(
+            SecurityPortfolioManager portfolio,
+            ISubscriptionDataConfigProvider subscriptionDataConfigProvider,
+            IOrderEventProvider orderEventProvider)
+        {
+            _maximumTradeableVolume = PortfolioStatistics.GetMaximumTradeableVolume();
+            _portfolio = portfolio;
+            _subscriptionDataConfigProvider = subscriptionDataConfigProvider;
+
+            _maximumSymbolCapacity = new Dictionary<Symbol, decimal>();
+            _historicalPortfolioCapacity = new RollingWindow<decimal>(30);
+            _previousInputTime = DateTime.MinValue;
+
+            orderEventProvider.NewOrderEvent += (sender, orderEvent) =>
+            {
+                var symbol = orderEvent.Symbol;
+
+                var configs = _subscriptionDataConfigProvider.GetSubscriptionDataConfigs(symbol);
+                if (configs.IsNullOrEmpty())
+                {
+                    Log.Error($"AssetsUnderManagementCapacityManager: Could not find {symbol} in Portfolio");
+                }
+
+                try
+                {
+                    var consolidator = default(IDataConsolidator);
+                    var security = _portfolio.Securities[symbol];
+                    BaseData lastData = security.Cache.GetData<TradeBar>();
+
+                    // For high resolution data, create a consolidator that will compute
+                    // the AUM Capacity after a 5-minute period bar is closed
+                    foreach (var config in configs.Where(x => x.Resolution < Resolution.Hour))
+                    {
+                        if (config.Type.IsAssignableFrom(typeof(TradeBar)))
+                        {
+                            consolidator = new TradeBarConsolidator(TimeSpan.FromMinutes(5));
+                        }
+
+                        if (config.Type.IsAssignableFrom(typeof(Tick)) &&
+                            config.TickType == TickType.Trade)
+                        {
+                            consolidator = new TickConsolidator(TimeSpan.FromMinutes(5));
+                            lastData = security.Cache.GetData<Tick>();
+                        }
+
+                        if (consolidator == null)
+                        {
+                            continue;
+                        }
+
+                        // Warm up the consolidator to mark the begging of the period
+                        if (lastData != null)
+                        {
+                            consolidator.Update(lastData);
+                        }
+
+                        config.Consolidators.Add(consolidator);
+                        consolidator.DataConsolidated += (s, data) =>  Update(s, data, orderEvent.OrderId);
+                        return;
+                    }
+
+                    // For low resolution, use the last data available
+                    if (lastData != null)
```

Comment:
Any concern is duplicating this data point in the consolidator? For example, having second SPY and daily SPY -- would we end up with the daily SPY config's consolidator updated each second? Should this be set up to limit the consolidators to one per unique symbol?